advntr/genome_analyzer.py

```diff
@@ -45,6 +50,96 @@ def print_genotype_in_bed_format(self, vntr_id, copy_numbers):
+        print('##INFO=<ID=VID,Number=1,Type=Integer,Description="VNTR ID"')
+        print('##INFO=<ID=RU,Number=1,Type=String,Description="Repeat motif"')
+        print('##INFO=<ID=RC,Number=1,Type=Integer,Description="Reference repeat unit count"')
+
+        print('##FORMAT=<ID=GT,Number=1,Type=String,Description="Genotype"')
+        print('##FORMAT=<ID=DP,Number=1,Type=Integer,Description="Read depth"')
+        print('##FORMAT=<ID=SR,Number=1,Type=Integer,Description="Spanning read count"')
+        print('##FORMAT=<ID=FR,Number=1,Type=Integer,Description="Flanking read count"')
+        print('##FORMAT=<ID=ML,Number=1,Type=Float,Description="Maximum likelihood"')
+
+        print("#CHROM\tPOS\tID\tREF\tALT\tQUAL\tFILTER\tINFO\tFORMAT\t" + self.input_file)
+
+    def print_genotype_in_vcf(self, vntr_id, genotype_result):
+        vntr = self.vntr_finder[vntr_id].reference_vntr
+
+        # CHROM
+        print(vntr.chromosome + "\t"),
+        # POS
+        start = vntr.start_point
+        end = start + vntr.get_length()
+        print(str(vntr.start_point) + "\t"),
+        # ID
+        id = '.'
+        print(id + "\t"),
+        # REF
+        ref = vntr.get_corresponding_region_in_ref()
+        print(ref + "\t"),
+        # ALT
+        consensus_seq = vntr.pattern  # TODO use consensus and confidence
+        GT = []
+        diff_count = 0
+        diff_copy_number = -1
+
+        if genotype_result.copy_numbers is None:
+            GT.append('.').append('.')
+        else:
+            for copy_number in genotype_result.copy_numbers:
+                if copy_number != vntr.estimated_repeats:
+                    diff_copy_number = copy_number
+                    diff_count += 1
+                    GT.append(diff_count)
+                else:
+                    GT.append(0)
+
+        if diff_count == 2:
+            print(consensus_seq * genotype_result.copy_numbers[0] + "," + consensus_seq * genotype_result.copy_numbers[1] + "\t"),
+        elif diff_count == 1:
+            print(consensus_seq * genotype_result.copy_numbers.index(diff_copy_number) + "\t"),
+        else:
+            print('.' + "\t"),
+
+        # QUAL
+        qual = '.'
+        print(qual + "\t"),
+        # FILTER
+        filter = '.'
+        print(filter + "\t"),
+
+        # INFO
+        advntr_vntr_ID = vntr_id  # ID
+        reference_repeat_pattern = vntr.pattern  # RU
+        reference_repeat_count = vntr.estimated_repeats  # RC
+
+        info_cols = "END=" + str(end) + ";VID=" + str(advntr_vntr_ID) + ";RU=" + reference_repeat_pattern +\
+                    ";RC=" + str(reference_repeat_count) + "\t"
+        print(info_cols),
+
+        # FORMAT
+        format_cols = "GT:DP:SR:FR:ML\t"
+        print(format_cols),
+
+        # GT
+        assert len(GT) == 2
+        print(str(GT[0]) + "/" + str(GT[1]) + ":"),  # '/' for unphased, '|' for phased.
+        # DP
+        stdout.write(str(genotype_result.recruited_reads_count) + ":"),
```

Comment:
Please modify these lines so that we use a single method for writing output and also make sure they work in both cases of writing to a file or stdout.